### PR TITLE
[vcpkg] Enable binary caching by default

### LINKED ISSUE #12285: Unable to read %LOCALAPPDATA% on Windows service accounts
After the merging of https://github.com/microsoft/vcpkg/pull/12091 (cc @ras0219), the instances of vcpkg running in our Windows CI exited reporting that `%LOCALAPPDATA%` cannot be read. This environment variable is undefined as the CI jobs are running on a Windows service account (running as SYSTEM).

As a hacky workaround, `$Env:LOCALAPPDATA = $Env:TEMP` solves the problem, but a proper fix would be much appreciated.

**Environment**
- OS: Windows Server 2019
- Compiler: MSVC 14.26
- Relevant environment variables (some have been redacted to `[...]`)
```
> Get-ChildItem Env:

Name                           Value
----                           -----
ALLUSERSPROFILE                C:\ProgramData
APPDATA                        C:\Windows\system32\config\systemprofile\AppData\Roaming
[...]
HOME                           C:\Windows\TEMP\drone-[...]\home\drone
HOMEPATH                       C:\Windows\TEMP\drone-[...]\home\drone
OS                             Windows_NT
Path                           C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;...
[...]
PSMODULEPATH                   WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;...
PUBLIC                         C:\Users\Public
SystemDrive                    C:
SystemRoot                     C:\Windows
TEMP                           C:\Windows\TEMP
TMP                            C:\Windows\TEMP
USERNAME                       [...]$
USERPROFILE                    C:\Windows\TEMP\drone-[...]\home\drone
VCPKG_ROOT                     C:\vcpkg
windir                         C:\Windows
```

**To reproduce**

1. Run `vcpkg install` on a Windows service account.
2. See error below:

    ```
    > vcpkg install

    Failed at [C:\vcpkg\toolsrc\src\vcpkg\userconfig.cpp(14)] with message:
    unable to read %LOCALAPPDATA%
    ```

**Expected behavior**

The command should succeed on service accounts. If *%LOCALAPPDATA%* is undefined, maybe it should fallback to one of the following directories:
- `C:\Windows\System32\config\systemprofile\AppData\Local`
- `C:\Windows\SysWOW64\config\systemprofile\AppData\Local`

### PR DESCRIPTION
**Pull request by ras0219**
Vcpkg will now accept an `x-` in front of any global (i.e. non-command-specific) command line parameters which are not already prefixed with `x-`. This has the effect that moving options from experimental to stabilized will not break existing users.

This additionally fixes #12285 by using `%APPDATA%/../Local` as a fallback for `%LOCALAPPDATA%`.

Drive by fix: error when passing multiple copies of single parameter arguments.

+@alexkaratarakis /* It's happening! */

### CODE REVIEW

**Comment by strega-nil** [comment]
Depends on #12351, afaict (or at least, that would make this PR much easier)

File: toolsrc/src/vcpkg/binarycaching.cpp [hunk 1]
```diff
@@ -660,6 +660,22 @@ ExpectedS<std::unique_ptr<IBinaryProvider>> vcpkg::create_binary_provider_from_c
...
+        static auto cachepath = System::get_platform_cache_home().then([](fs::path p) -> ExpectedS<fs::path> {
+            p /= fs::u8path("vcpkg/archives");
+            if (!p.is_absolute())
+            {
+                return {"default path was not absolute: " + p.u8string(), expected_right_tag};
+            }
+            else
+            {
+                return {std::move(p), expected_left_tag};
+            }
```
Comment (BillyONeal) [inline toolsrc/src/vcpkg/binarycaching.cpp:674]:
```suggestion
            if (p.is_absolute())
            {
                return {std::move(p), expected_left_tag};
            }
            else
            {
                return {"default path was not absolute: " + p.u8string(), expected_right_tag};
            }

```

**Review by BillyONeal — APPROVED** [review]

**Comment by alexkaratarakis** [comment]
:tada: 
Awesome! This will save a ridiculous amount of time :) .

**Pull request merged**